### PR TITLE
reduce (slightly) all About Pages load time using webpack code-splitting

### PR DESCRIPTION
**Pull request by petemill**
So it's only about 50ms - 100ms of saving, but another way to put it is 15% - 30% 😸 

Uses webpacks dynamic import capability, which will split the entry.js file in to multiple chunks, saving the browser having to retrieve and parse the parts of javascript that are not necessary for the specific about page that is to be loaded. Most useful for the 'new tab' page which is likely the most common extension page loaded, but before this PR, has to load and parse (although not run) all of the JS for about:preferences too...

I measured the performance by both:
- looking at the network panel 'load' time (after turning off the preference for wallpaper images on the 'new tab' page.
- adding `console.timeStamp('reactrendered')` as the third argument to `ReactDOM.render` in _js/about/entry.js_
   - then, running a dev tools Performance profile for page-load (using the refresh icon), and searching for the event 'reactrendered', noting the reported time the event occurred.

In a few tests I noticed that the average time to load and render the first react component for the 'new tab' page was:

- before this change: ~300ms - 350ms using a build (~600ms - 600ms on dev)
- after this change: ~250ms using a build (~420ms on dev)

Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Test Plan:
load `about:about` and click all of the about pages, checking they work as before

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -134,7 +134,9 @@
     "asar": "~0.13.0",
     "babel": "^6.1.18",
     "babel-core": "^6.3.15",
+    "babel-eslint": "^8.0.3",
     "babel-loader": "^7.1.1",
+    "babel-plugin-syntax-dynamic-import": "^6.18.0",
```
Comment (cezaraugusto) [inline package.json]: were the above deps reviewed by sec team?

File: js/about/entry.js [hunk 2]
```diff
@@ -17,64 +17,66 @@ ipc.send('request-language')
...
-    element = require('./extensions')
+    getElementOp = import('./extensions')
     break
   case 'about:history':
-    element = require('./history')
+    getElementOp = import('./history')
     break
   case 'about:newtab':
-    element = require('./newtab').AboutNewTab
+    getElementOp = import('./newtab').then(module => module.AboutNewTab)
```
Comment (cezaraugusto) [inline js/about/entry.js]: cool to see dynamics import for about pages

**Comment by cezaraugusto** [comment]
btw pls rebase too there's a conflict in package-lock

**Comment by bsclifton** [comment]
Security review created https://github.com/brave/internal/issues/162

Will wait for results before merging

**Comment by petemill** [comment]
security review passed, and branch is rebased on master

**Comment by bsclifton** [comment]
master eeba6b65ce71b9c3c5f79afc12837740c91f0291
0.21.x 5abcc9fa7b9621cec0ec7ca1192b9f5e71fdf966

**Pull request merged**